src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -186,12 +185,32 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
         /// 
         public static void PreloadAsmFromPath(string preloaderLocation, string asmLocation)
         {
-            if (string.IsNullOrEmpty(preloaderLocation) || !Directory.Exists(preloaderLocation))
+            // this will either be empty, the originally requested preloaderLocation, or a remapped location
+            // based on the old libG version number.
+            var preloaderLocationToLoad = "";
+
+            // if we can't find the preloader location directly as passed
+            // try converting it to a precise version location.
+            if (!Directory.Exists(preloaderLocation))
```

Comment:
What happens if `preloaderLocation` is empty or null?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.exists?view=netframework-4.7.2 apparently it will just return false

Reply:
relevant info:
```
The Exists method returns false if any error occurs while trying to determine if the specified file exists. This can occur in situations that raise exceptions such as passing a file name with invalid characters or too many characters, a failing or missing disk, or if the caller does not have permission to read the file.
```